The preflight playbooks should continue after error

brenton opened the pull request.

brenton committed:
The preflight playbooks should continue after error

brenton commented:
@sosiouxme, what do you think?

[test] just for fun.

sosiouxme commented:
Sure, the playbook should keep running after a play fails... but I'd still like it to indicate failure/success at the end, which it doesn't when you set ignore_errors. Not sure if we can capture that in a callback plugin or something.

brenton commented:
@sosiouxme, that makes sense.  Let me see if I can come up with a callback plugin for this.

In the meantime, I'm not sure [test] didn't trigger anything.

brenton committed:
HACK: Adding the stock default callback plugin

The following commit will contain the needed modifications.

brenton committed:
HACK: Correctly display the error count for preflight checks when ignore_errors is set.

I was struggling to get inheritance to work so this is a workaround.  In
reality this callback plugin is likely a placeholder for a simpler UI that we
will present once we have more checks in place.

brenton committed:
Adding a test for the preflight check callback plugin

brenton commented:
@sosiouxme PTAL!

sosiouxme commented on playbooks/byo/preflight/callback_plugins/default.py:
```diff
@@ -0,0 +1,311 @@
+# (c) 2012-2014, Michael DeHaan <michael.dehaan@gmail.com>
+#
+# This file is part of Ansible
```
Probably need to update the comments a bit here :) Otherwise looks good!
